conans/client/tools/scm.py

```diff
@@ -77,19 +77,25 @@ def _handle_url_pattern(self, scheme, url, user=None, password=None):
             return "{scheme}://{url}".format(scheme=scheme, url=url)
         elif scheme == "ssh" and self._password:
             self._output.warn("SCM password cannot be set for ssh url, ignoring parameter")
+        elif password and self._password:
+            self._output.warn("SCM password got from URL, ignoring 'password' parameter")
 
         if user and self._username:
             self._output.warn("SCM username got from URL, ignoring 'username' parameter")
-        if password and self._password:
-            self._output.warn("SCM password got from URL, ignoring 'password' parameter")
 
         the_user = user or self._username
         the_password = password or self._password
+
         if the_password and the_user and scheme != "ssh":
-            return "{scheme}://{user}:{password}@{url}".format(scheme=scheme, user=the_user,
-                                                               password=the_password, url=url)
+            return "{scheme}://{user}:{password}@{url}".format(scheme=scheme,
+                                                               user=quote_plus(the_user),
+                                                               password=quote_plus(the_password),
+                                                               url=url)
```

Comment:
other schemes like `file` or `git` should execute `quote_plus` too?

Reply:
`file` format is excluded in this return https://github.com/conan-io/conan/pull/8355/files#diff-dae658b6639940b7fd00c86a6ce57086f89c33ed5ae3a40a9a91f8ac49c12cd4R77, not sure for `git`

Reply:
Apparently it is not supported in the URL https://stackoverflow.com/questions/31801271/what-are-the-supported-git-url-formats and git protocol as no authentication support https://stackoverflow.com/a/33846897/9585455